react/hooks/useAppName.tsx

```diff
@@ -1,15 +1,19 @@
 import { useRuntime } from 'vtex.render-runtime'
+import { useAppVersionState } from '../components/AppVersionContext'
 
-export function useAppNameAndFile(major: string) {
+export function useAppNameAndFile() {
   const {
     route: { params },
   } = useRuntime()
 
   const { app, file } = params
-  const [appName] = app.split('@')
+  const appName = app ? app.split('@')[0] : null
   const fileName = file || 'README.md'
 
-  const finalAppName = `${appName}@${major}.x`
+  /* eslint-disable react-hooks/rules-of-hooks */
+  const major = appName ? useAppVersionState().major : null
+
+  const finalAppName = major ? `${appName}@${major}.x` : undefined
```

Comment:
I'm a bit split about this... not sure if this is the best way to do it.
Suggestions are welcome hahaha

Reply:
Looks alright